store/store.go

```diff
@@ -2223,7 +2236,7 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 					if channel == "" && a.Revision.Unset() {
 						channel = cur.TrackingChannel
 					}
-					refreshErrors[cur.Name] = translateSnapActionError("refresh", channel, res.Error.Code, res.Error.Message, res.Error.Extra.Releases)
+					refreshErrors[cur.InstanceName] = translateSnapActionError("refresh", channel, res.Error.Code, res.Error.Message, res.Error.Extra.Releases)
```

Comment:
is this change tested? also the other paths for install/download also need to use InstanceName, no?

Reply:
maybe download doesn't need to, but then maybe we should check that download InstanceName doesn't have a instance-key part, it seems code below assumes that but code above doesn't

Reply:
Added both a check to error out early if instance key is set in download and a fix for properly mapping errors for install actions. Also extended the tests a bit.